frontend/pages/hosts/ManageHostsPage/helpers.ts

```diff
@@ -0,0 +1,53 @@
+import { isString, isPlainObject, isEmpty, reduce, trim, union } from "lodash";
+
+export const getNextLocationPath = (
+  options = {
+    pathPrefix: "",
+    routeTemplate: "",
+    routeParams: {},
+    queryParams: {},
+  }
+): string => {
+  const pathPrefix = isString(options.pathPrefix) ? options.pathPrefix : "";
+  const routeTemplate =
+    (isString(options.routeTemplate) && options.routeTemplate) || "";
+  const routeParams = isPlainObject(options.routeParams)
```

Comment:
I'm curious to know what the alternative is for `options.routeParams` if it's not a plain object?

Reply:
I may have been a tad overzealous in trying to protect against runtime type errors here. The Lodash isPlainObject method here would intercept invalid arguments like strings or arrays that result in wacky return values. I wasn't sure how much I should rely on TypeScript to catch those since this currently gets imported into in a non-TypeScript component.